### ISSUE TITLE
508-defect-2 [COGNITION, SCREEN READER]: Buttons MUST have unique and accessible labels on the Military Service History page

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#18923]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I want to hear an accessible label like "Edit issues eligible for review" instead of "Edit" when the button has focus or virtual cursor.

## Details

There are several issues with this page which may ultimately be unsolvable from a remediation standpoint. I would recommend revisiting this page from a design lens. Issues are as follows:
- Edit buttons do not have unique and accessible labels. As such, they all currently announce "Edit" which doesn't tell the user _what_ they are editing.
- Inconsistency exists in legends when adding new military service as only the _last_ item in the stack has the legend of "New Service Period" despite the possibility of other _new_ service periods being added previously in the same session.
- The legend "New Service Period" does not change to "Edit Service Period" upon editing again _after_ saving which may confuse the user.

This will require a larger conversation or meeting to discuss solutions, so I'll only be documenting issues as opposed to recommendations.

For more details, view the screenshots attached on this ticket.

@1Copenut has worked on this page before in [ticket 1047](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1047) and may have some valuable historical context to add as well.

## Acceptance Criteria

- [ ] TBD

## Solution 

TBD

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

<img width="1792" alt="Screen Shot 2021-01-25 at 10 00 31 AM" src="https://user-images.githubusercontent.com/14154792/105726517-29231f00-5ef8-11eb-8878-f42e2336db1d.png">
<img width="1792" alt="Screen Shot 2021-01-25 at 10 09 47 AM" src="https://user-images.githubusercontent.com/14154792/105726526-2aece280-5ef8-11eb-9d17-c3996b923d58.png">
<img width="1337" alt="Screen Shot 2021-01-25 at 10 14 12 AM" src="https://user-images.githubusercontent.com/14154792/105726532-2c1e0f80-5ef8-11eb-86bb-272bb7782613.png">

This also exists on the review answers page where each edit button will announce "Edit military service button"
![all of these say edit military service button](https://user-images.githubusercontent.com/14154792/107548522-9e8f2080-6b83-11eb-8c09-248a995b456d.PNG)

**Comment by joshkimux** [comment]
Updating to must, following our accessibility ticketing guidelines
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#types-of-issues-1

**Comment by Mottie** [comment]
Everything should have `aria-label`s now!

**Comment by 1Copenut** [comment]
This looks 💯 @Mottie . I did a quick spot check of the aria-labels using Chrome's Inspect tool (it recently added an Accessibility bucket with accessible names!). All checked out and I'm ready to close this issue.
